app/src/main/java/org/fossasia/openevent/app/module/main/MainActivity.java

```diff
@@ -207,7 +220,28 @@ private void loadFragment(int navItemId) {
                 fragment = EventDashboardFragment.newInstance(eventId);
                 break;
         }
-        fragmentManager.beginTransaction().replace(R.id.fragment_container, fragment).commit();
+
+        if(!isAtHome){
+            fragmentManager.popBackStack("root_fragment",0);
+            isAtHome=true;
+        }
+        if(atHome){
+            root_fragment_id = lastSelectedNavItemId;
+            fragment_tag = "root_fragment";
+            fragmentManager.beginTransaction().replace(R.id.fragment_container, fragment).addToBackStack(fragment_tag).commit();
+            atHome=false;
+        }
+        else {
+            isAtHome = false;
+            if((lastSelectedNavItemId==R.id.nav_dashboard)||
+                (lastSelectedNavItemId==R.id.nav_events&&root_fragment_id==lastSelectedNavItemId)){
```

Comment:
Why such weird and static checks

Reply:
Suppose,we go from dashboard -> B -> dashboard. If we don't check this then what would happed is on the stack we will get dashboard->dashboard and then that would lead to improper back navigation.Same is for eventlist.